lib/remote/apilistener.cpp

```diff
@@ -261,6 +286,37 @@ void ApiListener::Start(bool runtimeCreated)
 	OnMasterChanged(true);
 }
 
+void ApiListener::RenewOwnCert()
+{
+	if (!Utility::PathExists(GetIcingaCADir() + "/ca.key")) {
+		return;
+	}
+
+	auto certPath (GetDefaultCertPath());
+	auto cert (GetX509Certificate(certPath));
+
+	if (IsCertUptodate(cert)) {
+		return;
+	}
+
+	Log(LogInformation, "ApiListener")
+		<< "Our certificate will expire soon, but we own the CA. Renewing.";
+
+	if (!(cert = RenewCert(cert))) {
```

Comment:
Can we please avoid assignments in if conditions without a good reason?

Reply:
For me, an overused meme does not qualify as a good reason.

Reply:
I'm all ears to your arguments.

Reply:
Functionally, assigning before the if and then checking is equivalent to how it's done here. So it's all about coding style.

I'd say that typically, if conditions don't assign variables, so it's somewhat surprising if one does. Yes, it can be done, but I think it's error prone and should be used with care. There's a reason why compilers issue warnings for `if (x = y)`, that's nothing you usually do, most likely you wanted to use `==` instead (sure, if you wanted to use `==` here, you'd have used `!=` in the first place).

Imagine you're interested where a variable is assigned and are skipping through all uses of that variable in a file. Which one is more likely to be overlooked?
* `cert = RenewCert(cert)`
* `if (!(cert = RenewCert(cert))) {`

In summary, as this is about style, it involves lots of personal preference. These are my 2 cents.